### PR TITLE
[DevEx-50] Replace local identity config with a new simplified module

### PR DESCRIPTION
**Pull request by Krusty93**
This PR add a new module under `infra/modules` path to simplify the creation and management of Azure Managed Identity federated with GitHub.

The new module asks for less variables and can manage both CI and CD identities.

The second part of the PR (`infra/identity/**`) replaces the current implementation with the new module. The complexity is extremely lower.
This can be considered as an example of implementation - teams want to adopt this approach would have a very similar code

### CODE REVIEW

File: infra/identity/prod/westeurope/main.tf [hunk 1]
```diff
@@ -19,6 +19,56 @@ provider "azurerm" {
...
+
+  prefix       = "dx"
+  env_short    = "p"
+  env          = "prod"
+  repositories = ["dx-typescript"]
+  domain       = "typescript"
+
+  continuos_integration = {
+    enable = true
+    roles = {
```
Comment (gunzip) [inline infra/identity/prod/westeurope/main.tf:40]: can we have this value with a default (within the module) in order to not have to specify these values here?

File: infra/identity/prod/westeurope/main.tf [hunk 2]
```diff
@@ -19,6 +19,56 @@ provider "azurerm" {
...
+        terraform-state-rg = [
+          "Storage Blob Data Contributor"
+        ]
+      }
+    }
+  }
+
+  continuos_delivery = {
+    enable = true
+    roles = {
```
Comment (gunzip) [inline infra/identity/prod/westeurope/main.tf:55]: ditto

File: infra/identity/prod/westeurope/main.tf [hunk 3]
```diff
@@ -19,6 +19,56 @@ provider "azurerm" {
...
+
+  tags = local.tags
+}
+
+module "federated_identities" {
+  source = "../../../modules/azure_identity_federation_github"
+
+  prefix       = "dx"
+  env_short    = "p"
+  env          = "prod"
```
Comment (gunzip) [inline infra/identity/prod/westeurope/main.tf:34]: can we have these values taken from a local file to the whole prod/westeurope directory?

File: infra/identity/prod/westeurope/main.tf [hunk 4]
```diff
@@ -19,6 +19,56 @@ provider "azurerm" {
...
-data "azurerm_subscription" "current" {}
+resource "azurerm_resource_group" "rg_identity" {
+  name     = "${local.project}-identity-rg"
+  location = local.location
+
+  tags = local.tags
+}
+
+module "federated_identities" {
+  source = "../../../modules/azure_identity_federation_github"
```
Comment (gunzip) [inline infra/identity/prod/westeurope/main.tf:30]: is this module supposed to be put into dx repo later?

Comment (Krusty93) [reply infra/identity/prod/westeurope/main.tf:30]: Yes, dx repo will have an exact copy of this code (module + internal adoption)

File: infra/identity/prod/westeurope/main.tf (hunk 3 above)
Comment (Krusty93) [reply infra/identity/prod/westeurope/main.tf:34]: Do you mean the locals.tf file? Yes we can

File: infra/identity/prod/westeurope/main.tf (hunk 1 above)
Comment (Krusty93) [reply infra/identity/prod/westeurope/main.tf]: Done

**Review by gunzip — APPROVED** [review]

**Pull request merged**